gossipsub_test.go

```diff
@@ -995,6 +996,83 @@ func TestGossipsubStarTopology(t *testing.T) {
 	}
 }
 
+// this tests overlay bootstrapping through px in Gossipsub v1.1, with addresses
+// exchanged in signed peer records.
+// we start with a star topology and rely on px through prune to build the mesh
+func TestGossipsubStarTopologyWithSignedPeerRecords(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	hosts := getNetHosts(t, ctx, 20)
+	psubs := getGossipsubs(ctx, hosts, WithPeerExchange(true))
+
+	// manually create signed peer records for each host and add them to the
+	// peerstores of the other hosts.
+	// this is necessary until the identify service is updated to exchange
+	// signed records
+	for i := range hosts {
```

Comment:
I think we should do this after the connect step and only add the peer records to the connected hosts.
This will allow us to test the actual exchange of records without having anything in the peer store and no routing to fall back on.

Reply:
Actually we don't have to add them anywhere other than the center of the star now that I think about it.

Reply:
done.